### PR TITLE
nix: add a fromGit argument so it can be overridden

### PR DESCRIPTION
**Pull request by catern**
There's no other way to override this attribute: override doesn't work
since it's not an argument of a callPackage call, and
overrideAttrs/overrideDerivation come in too late to change
things. Therefore, since nixUnstable is commented out, this change is
the only way to use fromGit and build from an unreleased git tag.

### CODE REVIEW

**Comment by matthewbauer** [comment]
Related: #37201

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -6,13 +6,14 @@
...
 , confDir ? "/etc"
+, fromGit ? false
 }:
 
 let
 
   sh = busybox-sandbox-shell;
 
-  common = { name, suffix ? "", src, fromGit ? false }: stdenv.mkDerivation rec {
+  common = { name, suffix ? "", src, fromGit ? fromGit }: stdenv.mkDerivation rec {
```
Comment (xeji) [inline pkgs/tools/package-management/nix/default.nix]: This creates an infinite recursion during eval. One of the two `fromGit` variables needs renaming.

**Comment by yegortimoshenko** [comment]
@GrahamcOfBorg eval

**Comment by yegortimoshenko** [comment]
@GrahamcOfBorg eval

**Comment by yegortimoshenko** [comment]
Eval fixed (at least locally), that said nixUnstable is not commented out anymore, so this probably doesn't make much sense in this case.

**Pull request closed without merging**